src/obstacle_avoidance/src/GoThroughHole.cpp

```diff
@@ -1,76 +1,55 @@
 /*
- * Created By: Gareth Ellis
- * Created On: July 16th, 2016
- * Description: An example node that subscribes to a topic publishing strings,
- *              and re-publishes everything it receives to another topic with
- *              a "!" at the end
+ * Created By: Ihsan Olawale
+ * Created On: February 16th, 2019
+ * Description: Uses LaserScan data of a setup of cones to move Olaf the robot
+ *              through the gap in the cones
  */
 
 #include <GoThroughHole.h>
 #include <laser_geometry/laser_geometry.h>
 #include <cmath>
 
-MyClass::MyClass(int argc, char **argv, std::string node_name) {
+ros::Rate loop_rate(10);
+
+GoThroughHole::GoThroughHole(int argc, char **argv, std::string node_name) {
     // Setup NodeHandles
     ros::init(argc, argv, node_name);
     ros::NodeHandle nh;
     ros::NodeHandle private_nh("~");
 
-    // Obtains character from the parameter server (or launch file), sets '!' as default
-    /*
-    std::string parameter_name = "my_node/character";
-    std::string default_character = "!";
-    SB_getParam(nh, parameter_name, suffix, default_character);
-    */
-
     // Setup Subscriber(s)
     std::string topic_to_subscribe_to = "scan";
     int refresh_rate = 10;
-    my_subscriber = nh.subscribe(topic_to_subscribe_to, refresh_rate, &MyClass::subscriberCallBack, this);
+    my_subscriber = nh.subscribe(topic_to_subscribe_to, refresh_rate, &GoThroughHole::subscriberCallBack, this);
 
     // Setup Publisher(s)
-    std::string topic = private_nh.resolveName("publish_topic");
+    std::string topic = private_nh.resolveName("move_olaf");
     uint32_t queue_size = 1;
-    my_publisher = private_nh.advertise<sensor_msgs::PointCloud>(topic, queue_size);
+    my_publisher = private_nh.advertise<geometry_msgs::Twist>(topic, queue_size);
+    
 }
 
-void MyClass::subscriberCallBack(const sensor_msgs::LaserScan::ConstPtr& msg) {
-    ROS_INFO("Received message");
-    /*
-    std::string input_string = msg->data.c_str();
-    std::string new_msg = addCharacterToString(input_string, suffix);
-    republishMsg(new_msg);
-    */
-    /*
-    ROS_INFO("\nThe ranges are: ");
-    for (int i = 0; i < msg->ranges.size(); i++) {
-        ROS_INFO("%f", msg->ranges[i]);
-    }
-    */
-    sensor_msgs::PointCloud cloud;
-    laser_geometry::LaserProjection projector_;
-    projector_.projectLaser(*msg, cloud); 
-    my_publisher.publish(cloud);
-    /*
-    ROS_INFO("The gaps and their intensities:");
-    std::vector<int> gaps = findGaps(*msg);
-    for (int i = 0; i < gaps.size(); i++) {
-        ROS_INFO("%i", gaps[i]);
-    }
-    */
+void GoThroughHole::subscriberCallBack(const sensor_msgs::LaserScan::ConstPtr& msg) {
+	stopOlaf();
+	loop_rate.sleep();
+	geometry_msgs::Point32 center = findHole(*msg);
+	if (std::sqrt(std::pow(center.x,2)+std::pow(center.y,2)) < 0.001) {
```

Comment:
What is going on in this if statement? You should add some comments explaining the logic, and also make sure not to use magic numbers. What does 0.001 correspond to? It'd be better to define a variable and make it equal to 0.001 so you could say...."some value < a_threshold_value". instead of "some value < 0.001". a_threshold_value is also not descriptive enough so choose a name that fits the naming.